src/tools/sssctl/sssctl_data.c

```diff
@@ -290,3 +291,60 @@ errno_t sssctl_cache_upgrade(struct sss_cmdline *cmdline,
 
     return EOK;
 }
+
+errno_t sssctl_cache_expire(struct sss_cmdline *cmdline,
+                            struct sss_tool_ctx *tool_ctx,
+                            void *pvt)
+{
+    errno_t ret;
+    char *cmd_args = NULL;
+    const char *cachecmd = SSS_CACHE;
+    char *cmd = NULL;
+    int i;
+
+    if (cmdline->argc == 0) {
+        fprintf(stderr, _("Missing option. \n"));
+
+        ret = sssctl_run_command(cachecmd);
+
+        if (ret != EOK) {
+            fprintf(stderr, _("Unable to expire cache objects\n"));
+        }
+
+        ret = EXIT_FAILURE;
+        goto done;
+    }
```

Comment:
I'm sorry that I noticed it earlier. Ant it take me quite long to review patches.
But is there a reason why we need to have "reports" to stderr here.

IMHO we should keep it simple and only call sss_cache. All logic is already there.
e.g `sss_cache` without arguments always return 1. And comment "Unable to expire cache objects" is confusing.
because sss_cache already reports info about usage; it didn't try to invalidate anything
```
[root@host ~]# sss_cache 
Usage: sss_cache [-?EUGNSAHR] [-?|--help] [--usage] [-E|--everything]
        [-u|--user=STRING] [-U|--users] [-g|--group=STRING] [-G|--groups]
        [-n|--netgroup=STRING] [-N|--netgroups] [-s|--service=STRING]
        [-S|--services] [-a|--autofs-map=STRING] [-A|--autofs-maps]
        [-h|--ssh-host=STRING] [-H|--ssh-hosts] [-r|--sudo-rule=STRING]
        [-R|--sudo-rules] [-d|--domain=STRING]
Please select at least one object to invalidate
[root@host ~]# echo $?
1
```